snaps/learn/tutorials/transaction-insights.md

```diff
@@ -28,54 +35,150 @@ Create a new Snap project using the
 
-You can resolve the issue by running the following command:
+Download
+[this `gas.svg` icon file](https://raw.githubusercontent.com/Montoya/gas-fee-snap/main/packages/snap/images/gas.svg)
+into that `ìmages` folder.  
+
+<details>
+  <summary>Icon attribution</summary>
+  <div>
+This is a free icon, "Gas" by Mello from the [Noun Project](https://thenounproject.com/browse/icons/term/gas/).
+ </div>
+</details>
+
+
+Your file structure should look like this:
+
+```text
+gas-estimation-snap/
+├─ packages/
+│  ├─ site/
+|  |  |- src/
+|  |  |  |- App.tsx
+|  |  ├─ package.json
+|  |  |- ...(react app content)
+|  |
+│  ├─ snap/
+|  |  ├─ images/
+|  |  |  |- gas.svg
+|  |  ├─ src/
+|  |  |  |- index.test.ts
+|  |  |  |- index.ts
+|  |  ├─ snap.manifest.json
+|  |  ├─ package.json
+|  |  |- ... (Snap content)
+├─ package.json
+├─ ... (other stuff)
+```
 
-```bash
-yarn run allow-scripts auto
+Open `packages/snap/snap.manifest.json` in a text editor. This file contains the main configuration 
+details for your Snap. Edit the `npm` object, within the `location` object, and add `iconPath` with 
+the value `"images/gas.svg"` to point to your new icon:
+
+```json title="snap.manifest.json"
+"location": {
+  "npm": {
+    "filePath": "dist/bundle.js",
+    "iconPath": "images/gas.svg",
+    "packageName": "snap",
+    "registry": "https://registry.npmjs.org/"
+  }
+}
 ```
 
-### 2. Enable transaction insights and the Ethereum provider
+Open `packages/snap/package.json` in a text editor. Edit the `files` array and reference the 
+`images/` folder:
 
-The default template Snap, such as the one in
-[Create a gas estimation Snap](gas-estimation.md),
-is configured to expose a JSON-RPC API with a simple hello command, which brings up a dialog box.
-In contrast, the Snap you're creating in this tutorial doesn't expose any API.
-Instead, it provides transaction insights directly in MetaMask's transaction confirmation window.
+```json title="package.json"
+"files": [
+  "dist/",
+  "images/",
+  "snap.manifest.json"
+],
+```
+
+#### 2.2. Update your Snap's wallet prompt
+
+It's important that user's understand what they're agreeing to when responding to wallet prompts. The 
+prompt uses the `proposedName` of the Snap, currently "Typescript Example" in our template.
 
-In particular, the Snap shows the user the percentage of gas fees they would pay for their transaction.
-It gets the current gas price by calling the
-[`eth_gasPrice`](/wallet/reference/eth_gasPrice) RPC
-method using the global Ethereum provider made available to Snaps.
+Open `packages/snap/snap.manifest.json` in a text editor.
+Edit the `"proposedName"` property within the metadata to provide a functional name such as "Get Fee %":
+
+```json title="snap.manifest.json"
+{
+  "version": "0.1.0",
+  "description": "An example Snap written in TypeScript.",
+  "proposedName": "Get Fee %",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/MetaMask/template-snap-monorepo.git"
+  },
```

Comment:
I had an idea for this -- would require tabbing between fragment and full file to clarify

Reply:
@alexandratran PTAL at the "fragments" vs "full page" example --> I agree that it can be confusing to look at fragments that don't looks as though they open/close properly (even though, in context, they do) WDYT does this help?

Reply:
OK, I see this comment here now. I think I prefer just the fragments. I find the tabs with the full file a bit confusing. If necessary, you can add an ellipses "..." before the closing bracket to denote additional text.